### ISSUE TITLE
CR for Azure Devops build-image.yml

### ISSUE THREAD

**Issue by joostmeijles** [Sitecore/docker-images#129]
I have the use-case that I only want to build Windows images using Azure Devops pipelines.

For this I propose to change the `build-image.yml`;
- introduce some conditionals for jobs
- add some variables for VM names, Azure subscription

In addition I would like to;
- stop VMs after build
- remove the assets drive mounting, caching locally seems okay and saves some complexity
- perform a `docker image prune` after each build

**Comment by jeanfrancoislarente** [comment]
The assets drive is used (by our team) since sometimes we work with assets that are not downloadable yet. So placing them on a file share is more convenient than figuring out where to download them from. It could be added as a conditional though.

Can you elaborate on "introduce some conditionals for jobs"?

the pipeline could also use some love to support building various versions, tags, etc.

I think the docker image prune thing could be scheduled "out of band" on the build server itself... Scheduled task to run at a given interval. The task would get run when the next time the build server spins up (if it has missed the schedule)... Issue is if you're troubleshooting builds, a prune wouldn't allow any sort of image cache use as it would start from scratch each time - which can be time consuming. Perhaps yet another flag in the build pipeline - can default to true but then set to false if you don't want to prune.

**Comment by joostmeijles** [comment]
> Can you elaborate on "introduce some conditionals for jobs"?
I now think adding a [condition](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/stages?view=azure-devops&tabs=yaml#dependencies) to the Linux stages is enough

> I think the docker image prune thing could be scheduled "out of band" on the build server itself
Agree, but there might be something to say for a clean build as well. A consecutive build now uses a previous builds side-effects.

**Comment by jeanfrancoislarente** [comment]
That's why I think the "prune" flag should be selectable... If you're trying to work through something and don't want a 3hr build, you can set prune to false but if you want a clean build then leave the default prune to true (or whatever we call it)

So sounds like 2 conditions / flags
Build Linux (true/false)
Prune after build true/false

That make sense?

**Comment by joostmeijles** [comment]
Created a PR with change proposal #136

**Comment by joostmeijles** [comment]
@jeanfrancoislarente the Docker login does not work for me. The line
```
$(container.registry.password) | docker login -u "$(container.registry.username)" --password-stdin $(container.registry.fullname)
```
gives me the following (secret variable related) error:
```
*** : The term '***' is not recognized as the name of a 
cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify 
that the path is correct and try again.
```

How did you get this to work?

PS. #136 contained some errors which I am currently fixing.

**Comment by joostmeijles** [comment]
Above is solved by adding some quotes, all fixes in #139

**Comment by joostmeijles** [comment]
Closed by #273
